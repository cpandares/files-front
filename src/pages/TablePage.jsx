import { Link } from "react-router-dom"


const TablePage = ({  data }) => {


  return (
    <table className="table table-striped table-bordered container mt-4 ">

        <thead>
            <tr>
                <th>FileName</th>
               
            </tr>
        </thead>

        <tbody>
          {
            data?.map( item =>(
                <tr key={item}>
                    <td>
                        <Link to={ `/${item}` } >
                            { item }
                        </Link>
                    </td>
                </tr>
            ) )
          }
        </tbody>
      
    </table>
  )
}

export default TablePage
