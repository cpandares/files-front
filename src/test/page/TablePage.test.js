import {  render } from '@testing-library/react';
import { MemoryRouter }   from 'react-router-dom';
import TablePage from '../../pages/TablePage';

const data = [
    "test1.csv",
    "test2.csv",
    "test3.csv",
    "test4.csv",
    "test5.csv",
    "test6.csv",
    "test9.csv"
]

describe("Pruebas en el <TablePage />", ()=>{

   test("Debe mostar el componente con los valores por defecto", ()=>{
    const  { container } = render(
        <MemoryRouter>
            <TablePage data={data}/>
        </MemoryRouter>
    )

    expect( container ).toMatchSnapshot()
   })
})