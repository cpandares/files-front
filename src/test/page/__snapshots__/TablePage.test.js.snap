// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <TablePage /> Debe mostar el componente con los valores por defecto 1`] = `
<div>
  <table
    class="table table-striped table-bordered container mt-4 "
  >
    <thead>
      <tr>
        <th>
          FileName
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            href="/test1.csv"
          >
            test1.csv
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/test2.csv"
          >
            test2.csv
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/test3.csv"
          >
            test3.csv
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/test4.csv"
          >
            test4.csv
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/test5.csv"
          >
            test5.csv
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/test6.csv"
          >
            test6.csv
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/test9.csv"
          >
            test9.csv
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
